test/test.js

```diff
@@ -376,6 +376,36 @@ suite('Response', function() {
       return json;
     })
   })
+  test('clone text response', function() {
+    var r = new Response('{"foo":"bar"}', {headers: {'content-type': 'application/json'}}),
+        clone = r.clone();
+    assert(clone.headers !== r.headers, true);
+    assert(clone.headers.get('content-type') === r.headers.get('content-type'), true);
+    return Promise.all([clone.json(), r.json()]).then(function(jsons){
+      assert(jsons[0].foo, 'bar');
+      assert(jsons[1].foo, 'bar');
+      return jsons;
+    })
+  })
+  test('clone blob response', function() {
+    return fetch('/binary').then(function(response) {
+      return Promise.all([response.clone().arrayBuffer(), response.arrayBuffer()]).then(function(bufs){
+        assert(bufs[0] instanceof ArrayBuffer, 'buf is an ArrayBuffer instance')
+        assert.equal(bufs[0].byteLength, 256, 'buf.byteLength is correct')
+        var view0 = new Uint8Array(bufs[0])
+        for (var i = 0; i < 256; i++) {
+          assert.equal(view0[i], i)
+        }
+        assert(bufs[1] instanceof ArrayBuffer, 'buf is an ArrayBuffer instance')
```

Comment:
You're repeating identical logic as for `bufs[0]` here. How about iterating through both `bufs` in a loop and applying the same block of assertions?